src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.GenericLookups.cs

```diff
@@ -146,8 +146,11 @@ public GenericLookupResult Type(TypeDesc type)
             public GenericLookupResult UnwrapNullableType(TypeDesc type)
             {
                 // An actual unwrap nullable lookup is only required if the type is exactly a runtime 
-                // determined type associated with System.__UniversalCanon itself
-                if (type.IsRuntimeDeterminedType && ((RuntimeDeterminedType)type).CanonicalType.IsCanonicalDefinitionType(CanonicalFormKind.Universal))
+                // determined type associated with System.__UniversalCanon itself, or if it's
+                // a runtime determined instance of Nullable.
+                if (type.IsRuntimeDeterminedType && (
```

Comment:
Only doing this for `__UniversalCanon` is actually a bug that could be hit on Project X.